src/dos/dos_files.cpp

```diff
@@ -50,6 +50,136 @@ DOS_File* Files[DOS_FILES] = {};
 // the drive manager class.
 std::array<DOS_Drive*, DOS_DRIVES> Drives = {};
 
+enum class FileSharingMode
+{
+	Compatibility,
+	DenyReadWrite,
+	DenyWrite,
+	DenyRead,
+	DenyNone
+};
+
+// https://stanislavs.org/helppc/int_21-3d.html
+union FileFlagBits
+{
+	uint8_t data = 0;
+	bit_view<0, 3> access_mode;
+	bit_view<4, 3> sharing_mode;
+};
+
+struct FileAccessMode
+{
+	bool read = false;
+	bool write = false;
+};
+
+struct FileOpenFlags
+{
+	FileSharingMode sharing = FileSharingMode::Compatibility;
+	FileAccessMode access = {};
+};
+
+static FileOpenFlags parse_file_flags(const uint8_t flags)
+{
+	FileOpenFlags ret = {};
+
+	const FileFlagBits flag_bits = {flags};
+
+	switch (flag_bits.access_mode) {
+		case OPEN_READ:
+		case OPEN_READ_NO_MOD:
+			ret.access.read = true;
+			ret.access.write = false;
+			break;
+		case OPEN_WRITE:
+			ret.access.read = false;
+			ret.access.write = true;
+			break;
+		case OPEN_READWRITE:
+			ret.access.read = true;
+			ret.access.write = true;
+			break;
+		// Assume read/write access for the purpose of file locking checks
+		// This should throw an error when the file actually tries to open
+		default:
```

Comment:
Comment should go inside the `default` branch & fix indentation.

Reply:
Fixed